.github/workflows/zfs-tests-sanity.yml

```diff
@@ -6,32 +6,26 @@ on:
 
 jobs:
   tests:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```

Comment:
Let's leave one of runners on latest to get problems as fast as we can. We may add an explicit comment about the reason.

Reply:
I'm sorry, I can't quite see your point here :(

`-latest` are rolling tags pointing to the most recent release, e. g. `ubuntu-latest -> ubuntu-20.04`. Ubuntu LTS is released on 2 year cadence. Using `-latest` here would silently update the runner to `-22.04`, which, most likely, would lead to CI failures. Potential resolutions: either downgrade to `-20.04` or fix broken workflows in hurry.

Reply:
> or fix broken workflows in hurry.

Yep, in addition we have a runner which tries ZFS with kernel's master `buildbot/Kernel.org Built-in x86_64`, so failed CI tells us about changes.

Reply:
It's not about kernel, it's about user land: compiler, linker, other tools. From my personal experience I can tell that the build process behaves differently e. g. on my Debian testing due to different versions of gcc, mandoc and shellcheck.

The purpose of `buildbot/Kernel.org Built-in x86_64` is to check how well OpenZFS master integrates with Linux master. Taking rolling versions (masters) makes a lot of sense for that purpose.
ZTS purpose is to check if a particular OpenZFS revision break the regression suite. Adding a variable to that process in a form of changing toolset seems unlikely to bear fruits.

If the concern is the code base ability to build I'd propose to set up similar integration job as it is for kernel now with Ubuntu rolling or Debian testing/unstable. It would be nice to be alarmed about changes in tools in advance; also developers using rolling Linux distros / FreeBSD current would have easier life.

Reply:
> If the concern is the code base ability to build I'd propose to set up similar integration job as it is for kernel now with Ubuntu rolling or Debian testing/unstable

We use github actions here, it gives us limited distros from the box https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners , so ubuntu-latest is the only easy way here.

Let's not harden CI pretty hard, it's a great way to get problems as fast as we can, and it is the purpose here.